packages/backend/src/core/DiscoveryWatcher.ts

```diff
@@ -29,6 +29,9 @@ export class DiscoveryWatcher {
     this.taskQueue = new TaskQueue(
```

Comment:
I'm thinking if we could have a method like `TaskQueue.for(this, ...constructorArgs)` Then we could get the ID from `constructor.name` like in logger. WYT?